lib/services/responses.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+
+const bluebird = require('bluebird'),
+  _includes = require('lodash/includes'),
+  _assign = require('lodash/assign');
+
+/**
+ * Handles errors in the standard/generic way
+ * @param {Object} res
+ * @returns {Function}
+ */
+function handleError(res) {
+  return ({ code, stack }) => {
+    if (code) {
+      res.status(code).send(stack);
+    } else {
+      res.send(stack);
+    }
+  };
+}
+
+/**
+ * Responds with JSON and capture
+ * Captures and hides appropriate errors.
+ * These return JSON always, because these endpoints are JSON-only.
+ * @param {Function} fn
+ * @param {Object} res
+ * @returns {Promise}
+ */
+function expectJSON(fn, res) {
+  return bluebird.try(fn).then((result) => res.json(result)).catch(handleError(res));
+}
+
+/**
+ * Creates a method validation middleware.
+ * @param {{allow: [String]}} options
+ * @returns {Function}
+ */
+function methodNotAllowed(options) {
```

Comment:
I find this name/signature confusing. Should be something like `methodAllowed(...method)`

Reply:
it makes more sense if you look at [where it's used](https://github.com/clay/amphora-schedule/pull/1/files#diff-d399259ffb3aad613261f7c9d55539daR39). The check for allowed methods is misleading here, but realistically it's only called on a router method thats explicitly not allowed. the only reason the allowed methods are passed here is for posterity in the response